lib/reek/ast/sexp_extensions/constant_assignment.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+module Reek
+  module AST
+    module SexpExtensions
+      # Utility methods for :casgn nodes.
+      module CasgnNode
+        def assignment
+          children[2]
+        end
+
+        def class_creation?
+          assignment.respond_to?(:name) &&
```

Comment:
This is a brute way to avoid exceptions while calling `class_creation?` on something like:
```
s(:casgn, nil, :Iterator,
  s(:block,
    s(:send,
      s(:const, nil, :Struct), :new,
      s(:sym, :exp),
      s(:sym, :depth)),
    s(:args),
    s(:begin,
      s(:send, nil, :include,
        s(:const, nil, :Comparable)),
      s(:def, :<=>,
        s(:args,
          s(:arg, :other)),
        s(:send,
          s(:send, nil, :depth), :<=>,
          s(:send,
            s(:lvar, :other), :depth))))))
```

This is an example of class_creation:
```
s(:casgn, nil, :Dummy,
  s(:send,
    s(:const, nil, :Class), :new,
    s(:const, nil, :Array)))
```

Can you help me with something more robust?